File: py/Task374.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):


class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left, right = 1, n
        while left <= right:
            mid = left + (right-left >> 1)
            result = guess(mid)
            if result == 0:
                return mid
            elif result == 1:
                left = mid+1
            else:
                right = mid-1
        return -1

File: go/task374/task374.go
package task374

// binary search
func guessNumber(n int) int {
	left, right := 1, n
	for left <= right {
		mid := left + (right-left)>>1
		if guess(mid) == 0 {
			return mid
		} else if guess(mid) < 0 {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}
	return -1
}
